chapters/part3/fairness.md

```diff
@@ -0,0 +1,731 @@
+
+`AssociationWithoutGroundTruth` can be used like any other `Metric` in the library, and it is supported in distributed settings. Here is an example code snippet for using this metric with your own models:
+
+```python
+from allennlp.models.model import Model
+from allennlp.fairness.bias_metrics import AssociationWithoutGroundTruth
+
+class YourModel(Model):
+    
+    def __init__(self, *args, **kwargs):
+        ...
+        # Initialize AssociationWithoutGroundTruth metric object
+        self._npmixy = AssociationWithoutGroundTruth()
+        ...
+
+    def forward(self, *args, **kwargs):
+        ...
+        # Accumulate metric over batches
+        self._npmixy(predicted_labels, protected_variable_labels)
+        ...
+
+model = YourModel(...)
+...
+# Get final values of metric after all batches have been processed
+print(model._npmixy.get_metric())
+```
+</exercise>
+
+<exercise id="5" title="Applying Bias Mitigation to Large, Contextual Language Models">
+
+The previous sections detail bias mitigation and direction methods and bias metrics. However, you might be wondering, how can I apply these tools directly from a config file? Additionally, you probably noticed that they are intended for <em>static</em> word embeddings, not the large, contextual language models you're likely interested in using.
+
+Fortunately, `allennlp.fairness` provides a `Model` wrapper called [`BiasMitigatorApplicator`](http://docs.allennlp.org/main/api/fairness/bias_direction_wrappers/) to mitigate biases in contextual embeddings during finetuning on a downstream task and test time, based on Dev, S., Li, T., Phillips, J.M., & Srikumar, V. (2020). [On Measuring and Mitigating Biased Inferences of Word Embeddings](https://api.semanticscholar.org/CorpusID:201670701). ArXiv, abs/1908.09369. In this section, we detail how to apply `BiasMitigatorApplicator` to reduce binary gender bias in a large RoBERTA model for SNLI.
+
+1. The first step is to obtain the pretained model for which you would like to reduce binary gender bias. In this case, we will use [AllenNLP's public large RoBERTA model for SNLI](https://storage.googleapis.com/allennlp-public-models/snli-roberta.2021-03-11.tar.gz).
+
+2. `BiasMitigatorApplicator` works by applying a bias mitigator of your choice directly after the static embedding layer of the pretrained RoBERTA model during finetuning and evaluation. This reduces the model's ability to propagate gender biases across words in input sentences through contextualization.
```

Comment:
To me, this is the real magic, because this is the thing that allows people to use your work without having to understand it all. That's the hallmark of a good abstraction.

For that same reason, I think it's important to have a paragraph about the limitations of `BiasMitigatorApplicator`. Does it work with any model that uses transformers? Will it work equally well on all of them? Can we characterize the performance we can get out of it, both in terms of how well it mitigates bias, and in terms of the final model performance?

Reply:
This is a great point! Just added a paragraph.